Update React Native guide for push notifications plugin (#3478)
ably/docs: src/pages/docs/push/getting-started/react-native.mdx

- `key`: Your Ably API key.
- `clientId`: A unique identifier for this client.
- `plugins`: The push plugin created with `ReactNativePush.create()`. Your app supplies the two things the SDK cannot provide itself on React Native: persistent storage (any implementation of the `@react-native-async-storage/async-storage` interface) and a `requestToken` callback that returns a push token. Return `transportType: 'fcm'` for a Firebase Cloud Messaging registration token, or `transportType: 'apns'` for a raw APNs device token obtained from `messaging().getAPNSToken()`.

`AblyProvider` makes the Ably client available to all child components via React context. `ChannelProvider` scopes child components to `my-first-push-channel`. `useChannel` subscribes to realtime messages published on the channel.

## Step 2: Set up push notifications <a id="step-2"/>

> Activate push notifications with [`push.activate()`](/docs/api/realtime-sdk/push#activate). Requesting notification permission from the user is your app's responsibility and must happen before activation. Add the following inside `PushScreen`:

Reviewer: ```suggestion
Activate push notifications with [`push.activate()`](/docs/api/realtime-sdk/push#activate). Your app must request notification permission from the user before activation. Add the following inside `PushScreen`:
```

    }

    await messaging().registerDeviceForRemoteMessages(); // Required to receive push notifications on iOS, no-op on Android
    const fcmToken = await messaging().getToken();
    await saveDeviceRegistration(fcmToken);

    tokenRefreshUnsubscribeRef.current?.();

>     tokenRefreshUnsubscribeRef.current = messaging().onTokenRefresh(async newToken => {

Reviewer 2: how is onTokenRefresh supported with the new plugin? is it worth adding a note somewhere about what happens with tokens are rotated now?

Author: fixed
